data_structures/src/error.rs

```diff
@@ -276,7 +276,7 @@ pub enum TransactionError {
     DataRequestWeightLimitExceeded {
         weight: u32,
         max_weight: u32,
-        dr_output: DataRequestOutput,
+        dr_output: Box<DataRequestOutput>,
```

Comment:
Very interesting. Do you know why are they pushing for using `Box`es liberally like this?

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#result_large_err

Basically, without the box, the `TransactionError` enum is too big, and that hurts performance a bit. We could have changed all the uses of `Result<(), TransactionError>` to `Result<(), Box<TransactionError>>` but in this case I preferred to box just this one large field.